app/src/main/java/ch/sdp/vibester/activity/LyricsBelongGameActivity.kt

```diff
@@ -120,10 +121,7 @@ class LyricsBelongGameActivity : AppCompatActivity() {
                     val result = response.body()
                     if (response.isSuccessful && result != null) {
                         lyrics = result.lyrics.toString().replace(",", "")
-                        checkLyrics(
-                            speechInput,
-                            lyrics
-                        ) // be sure the lyrics is ready when checking
+                        checkLyrics( speechInput, lyrics, gameManager) // be sure the lyrics is ready when checking
                     } else {
                         findViewById<TextView>(R.id.lyricMatchResult).text =
                             "No lyrics found, try another song"
```

Comment:
the progress bar should be stopped if lyrics are not found

Reply:
Thanks for spotting this out, I will replace all the use of the next button to the progress bar and redesign the switch logic in my next PR.